### PR TITLE
feat(AHWR-319): Event processing fails because of clashing primary keys

### PR DESCRIPTION
**Pull request by omega1119**
extended rowKey to include the eventType and check if duplicates exists as well as testing

### CODE REVIEW

File: ffc-ahwr-event/event/event.js [hunk 1]
```diff
@@ -11,7 +12,7 @@ const saveEvent = async (context, event) => {
   const timespan = new Date(raisedEvent.action.raisedOn).getTime()
 
   const partitionKey = `${sbi}`
-  let rowKey = `${partitionKey}_${timespan}`
+  let rowKey = buildRowKey(partitionKey, timespan, eventType)
```
Comment (mattyod) [inline ffc-ahwr-event/event/event.js:15]: Hate to say but it feels a bit unnecessary to abstract this into a function. A template string tells me then and there what the structure of the rowKey is.

Comment (omega1119) [reply ffc-ahwr-event/event/event.js:15]: I abstracted it that way for the unit test

File: test/unit/event/event.test.js [hunk 2]
```diff
@@ -58,4 +59,31 @@ describe('Event function', () => {
     expect(queryEntities).toHaveBeenCalledTimes(1)
     expect(mockContext.bindings).toHaveProperty('tableBinding')
   })
+
+  test('should construct the rowKey correctly', () => {
```
[the remark this answers is not in the document]
Comment (omega1119) [reply test/unit/event/event.test.js]: I have created that test now

File: ffc-ahwr-event/event/event.js (hunk 1 above)
Comment (mattyod) [reply ffc-ahwr-event/event/event.js:15]: but if you didn't abstract it away we wouldn't have to write and maintain tests for it 😄  The outcome is already tested in `event.test.js`.

Comment (aaroncarroll82) [reply ffc-ahwr-event/event/event.js:15]: Agree with Matt, the separate function doesn't really give us anything. The 1 liner in this file, plus the unit test in test/unit/event/event.test.js should be sufficient

**Pull request merged**